src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -849,7 +850,292 @@ def coefficient_matrix(self, sparse=True):
 
         R = self.ring()
         A, v = self.coefficients_monomials(sparse=sparse)
-        return A, matrix(R,len(v),1,v)
+        return A, matrix(R, len(v), 1, v)
+
+    def macaulay_matrix(self, degree,
+                        homogeneous=False,
+                        variables=None,
+                        return_indices=False,
+                        remove_zero=False,
+                        reverse_column_order=False,
+                        row_order=None):
+        r"""
+        Return the Macaulay matrix of degree ``degree`` for this sequence
+        of polynomials.
+
+        INPUT:
+
+        - ``homogeneous`` -- boolean (default: ``False``);
+          when ``False``, the polynomials in the sequence do not need to be homogeneous
+          the rows of the Macaulay matrix correspond to all possible products
+          between a polynomial from the sequence and monomials of the polynomial
+          ring up to degree ``degree``;
+          when ``True``, all polynomials in the sequence must be homogeneous.
```

Comment:
```suggestion
          when ``True``, all polynomials in the sequence must be homogeneous, and
```
(or if keeping the dot ".", add a capital letter to next word)